src/test_task.c

```diff
@@ -134,17 +137,76 @@ test_restraint_task_free (void)
     restraint_task_free (task);
 }
 
+static void
+test_restraint_task_get_offset (void)
+{
+    Task        *task;
+    const gchar *path;
+    goffset     *offset;
+
+    path = "/some/random/path.log";
+
+    task = restraint_task_new ();
+
+    g_assert_nonnull (task);
+
+    offset = restraint_task_get_offset (task, path);
+
+    assert_offset (task->offsets, path, 0);
+
+    g_assert_true (offset == restraint_task_get_offset (task, path));
+
+    restraint_task_free (task);
+}
+
+static void
+test_task_config_set_offset (void)
+{
+    Task               *task;
+    const gchar        *path;
+    g_autofree gchar   *config_file;
+    g_autoptr (GError)  err = NULL;
+    goffset             value;
+
+    config_file = g_build_filename (tmp_test_dir, "config.conf", NULL);
+    path = "/some/random/path.log";
+    value = 42;
+
+    task = restraint_task_new ();
+
+    g_assert_nonnull (task);
+
+    task->task_id = g_strdup_printf ("%" G_GINT64_FORMAT, g_get_real_time ());
+
+    g_assert_true (task_config_set_offset (config_file, task, path, value, &err));
+    g_assert_no_error (err);
+
```

Comment:
I think it would be fruitful to get the offset and verify it contains value of 42.

Reply:
Your change isn't complete.  All you are checking is that value is initialized to 42 which I can already see that it is.  My comment was to verify that set_offset achieve the goal of setting the value correctly in the config file. This would require a 'get_offset' operation too.

Reply:
> All you are checking is that value is initialized to 42 which I can already see that it is.

I don't follow. Where am I checking that?

>  to verify that set_offset achieve the goal of setting the value correctly in the config file

That's what `assert_key_file_offset` is doing.

> This would require a 'get_offset' operation too.

There is no get_offset from config file defined. And note that `assert_offset` is not relevant here.

Reply:
yes I see it now